src/rpc/rawtransaction.cpp

```diff
@@ -916,13 +917,17 @@ static UniValue testmempoolaccept(const JSONRPCRequest& request)
 
     TxValidationState state;
     bool test_accept_res;
+    CAmount fee; // To return transaction fee from AcceptToMemoryPool
     {
         LOCK(cs_main);
         test_accept_res = AcceptToMemoryPool(mempool, state, std::move(tx),
-            nullptr /* plTxnReplaced */, false /* bypass_limits */, max_raw_tx_fee, /* test_accept */ true);
+            nullptr /* plTxnReplaced */, false /* bypass_limits */, max_raw_tx_fee, /* test_accept */ true, &fee);
     }
     result_0.pushKV("allowed", test_accept_res);
-    if (!test_accept_res) {
+    if (test_accept_res) {
+        // Push tx fee, only if test_accept_res == true
```

Comment:
No need for a comment to say what the code is doing. If you want to comment here, it should be to explain _why_ the code is doing this, eg "if mempool acceptance failed, the returned fee may not be accurate, so only return it to the caller if the tx would be accepted." or similar.

Reply:
Updated this comment to better express the intent.